### PR TITLE
Member search for learndash open course

### PR DESCRIPTION
**Pull request by asaquzzamanbb**
### All Submissions:

* [ ] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [ ] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes https://github.com/buddyboss/buddyboss-app/issues/1919

### How to test the changes in this Pull Request:

When any learndash course access mode is set as 'open' then its enrolled user does not appear in members search result. 

`The Course price type is set to "open". This means ALL are automatically enrolled`

https://www.loom.com/share/ff93e1ed0d854d1dbd58bc32854d4d3b

### Proof Screenshots or Video

https://www.loom.com/share/1353810069c646be8982dcf083ef5a5b

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### Related Platform API PR

https://github.com/buddyboss/buddyboss-platform-api/pull/246

### CODE REVIEW

**Comment by dipskakadiya** [comment]
@chetansatasiya @KartikSuthar Can you review this PR? I don't have much context for this in platform.

File: src/bp-integrations/learndash/learndash/Hooks.php [hunk 1]
```diff
@@ -45,6 +45,32 @@ public function init() {
...
+	 * @uses learndash_get_setting() Get learndash individual course settings.
+	 * 
+	 * @return bool
+	 */
+	public function checkLearndashCourseFilter( $has_access, $field ) {
+		if ( 'field_learndash_courses' === $field->code ) {
+			$course_settings = learndash_get_setting( $field->value );
+			
+			// Skip the leardash course field from member search when course price type or course mode is 'open'.
+			if ( 'open' == $course_settings['course_price_type'] ) {
```
Comment (chetansatasiya) [inline src/bp-integrations/learndash/learndash/Hooks.php:68]:
```suggestion
			if ( 'open' === $course_settings['course_price_type'] ) {
```
Check for a strict comparision

File: src/bp-integrations/learndash/learndash/Hooks.php [hunk 2]
```diff
@@ -45,6 +45,32 @@ public function init() {
 		add_action( 'update_post_meta', array( $this, 'checkLearndashCourseUpdateMeta' ), 10, 4 );
 		add_action( 'added_post_meta', array( $this, 'checkLearndashCourseUpdateMeta' ), 10, 4 );
 		add_action( 'deleted_post_meta', array( $this, 'checkLearndashCourseDeleteMeta' ), 10, 4 );
+		add_filter( 'bp_ps_field_can_filter', array( $this, 'checkLearndashCourseFilter' ), 10, 2 );
+	}
+
+	/**
+	 * Skip the leardash course field from member search when course price type or course mode is 'open'.
+	 * 
+	 * @since BuddyBoss 1.5.9
```
Comment (chetansatasiya) [inline src/bp-integrations/learndash/learndash/Hooks.php:54]:
```suggestion
	 * @since BuddyBoss 1.6.0
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need to make proper since tag and strict comparision.

**Review by chetansatasiya — APPROVED** [review]

**Comment by mohit-buddyboss** [comment]
@KartikSuthar @chetansatasiya @asaquzzamanbb 

To this PR all members on directory page must appear correctly but on dev branch members count on directory page is not correct because of this PR: https://github.com/buddyboss/buddyboss-platform/pull/2245
@asaquzzamanbb resolved issue mentioned on above PR by Sumit. Now this PR needs review. @KartikSuthar can you please review this PR: PR: https://github.com/buddyboss/buddyboss-platform/pull/2245


cc @sourabhmatolia

**Pull request merged**